cradle-cassandra/src/main/java/com/exactpro/cradle/cassandra/ReadThroughBookCache.java

```diff
@@ -95,16 +95,18 @@ public Collection<PageInfo> loadPageInfo(BookId bookId, Instant start, Instant e
         LocalTime startTime = start != null ? toLocalTime(start) : LocalTime.MIN;
         LocalDate endDate = end != null ? toLocalDate(end) : LocalDate.MAX;
         LocalTime endTime = end != null ? toLocalTime(end) : LocalTime.MAX;
-        for (PageEntity pageEntity : operators.getPageOperator().get(
+        for (PageEntity pageEntity : operators.getPageOperator().getByEnd(
```

Comment:
page.endDate and page.endTime aren't partition or cluster key. We have two cases to avoid loading a lot data
1) use `ALLOW FILTERING` - it is bad practice and increase cassandra load
2) create index for these column - we can experiment with that but I won't like do that in this PR

Reply:
I think about condition in loop

Reply:
Hm... the `public Collection<PageInfo> loadPageInfo(BookId bookId, Instant start, Instant end, boolean loadRemoved)` method isn't a part of public API. now the method is used in in our page cache where it isn't interrupted.
Maybe change visibility to package private?

Reply:
Hi, @Nikita-Smirnov-Exactpro. But what about this loop? Shouldn't we stop iterating once the page start time is lower than `start` parameter?

> Maybe change visibility to package private?

I have no preference about the visibiilty

Reply:
Hi, @Nikita-Smirnov-Exactpro. But what about this loop? Shouldn't we stop iterating once the page start time is lower than start parameter?

No, cassandea request returns page from past to future and we should filter all pages where page end < start parameter.
We are interested in all pages covered requested interval [start, end ) 
| ... start ... | ... end
| ... (start |) ... end
| ... start ... end ... |
 start ... | ... end ... |

Reply:
Ah, okay. Maybe we should modify the query (create a new one) that will return pages in reversed order? It should be simpler for us in this case - we won't iterate over all pages from the beginning

Reply:
@OptimumCode implemented method to load all pages before in descending order